cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -35,10 +37,37 @@
    * Represents all known trigger types in REST requests/responses.
    */
   public enum Type {
-    TIME,
-    PARTITION,
-    STREAM_SIZE,
-    PROGRAM_STATUS
+    TIME("time"),
+    PARTITION("partition"),
+    STREAM_SIZE("stream-size"),
+    PROGRAM_STATUS("program-status");
+
+    private static final Map<String, Type> CATEGORY_MAP;
+
+    static {
+      CATEGORY_MAP = new HashMap<>();
+      for (Type type : Type.values()) {
+        CATEGORY_MAP.put(type.getCategoryName(), type);
+      }
+    }
+
+    private final String cateogryName;
+
+    Type(String cateogryName) {
+      this.cateogryName = cateogryName;
+    }
+
+    public String getCategoryName() {
+      return cateogryName;
+    }
+
+    public static Type valueOfCategoryName(String categoryName) {
```

Comment:
So is the only use case for this to take in lowercase (or case insensitive) trigger types and make sure they return the actual type? If so, then I think using the `GSON` object already declared in `ProgramLifecycleHttpHandler` is a better solution than this categoryName map - it parses a lot more than just the "categoryName":

```
GSON.fromJson("program_status", ProtoTrigger.Type.class)
GSON.fromJson("tiMe", ProtoTrigger.Type.class)
GSON.fromJson("program_STATUS", ProtoTrigger.Type.class)
```

All of those get parsed to their appropriate Trigger types.